### PR TITLE
Use SignedPayload[T] and JsonSignedPayload

### PR DESCRIPTION
**Pull request by simao**
Depends on https://github.com/advancedtelematic/ota-tuf/pull/190

### CODE REVIEW

File: src/test/scala/com/advancedtelematic/director/http/AdminResourceSpec.scala [hunk 1]
```diff
@@ -102,7 +102,7 @@ trait AdminResourceSpec extends DirectorSpec with KeyGenerators with DeviceRegis
     }
   }
 
-  testWithNamespace("devices/id gives a list of ecuresponses") { implicit ns =>
+  testWithNamespace("XXX devices/id gives a list of ecuresponses") { implicit ns =>
```
Comment (jochenschneider) [inline src/test/scala/com/advancedtelematic/director/http/AdminResourceSpec.scala]: ?

**Comment by jerrytrieu** [comment]
I will rebase and tag this commit.

**Comment by jochenschneider** [comment]
Cool, thanks.

File: build.sbt [hunk 2]
```diff
@@ -14,7 +14,7 @@ libraryDependencies ++= {
   val akkaHttpV = "10.0.11"
   val scalaTestV = "3.0.0"
   val bouncyCastleV = "1.57"
-  val tufV = "0.4.0-26-gbd9fd3c"
+  val tufV = "0.4.0-40-ga8148f0-SNAPSHOT"
```
Comment (simao) [inline build.sbt]: this still needs to be updated

**Pull request merged**